gobblin-temporal/src/test/java/org/apache/gobblin/temporal/joblauncher/GobblinTemporalJobLauncherTest.java

```diff
@@ -0,0 +1,185 @@
+      super(jobProperties, appWorkDir, new ArrayList<>(), new ConcurrentHashMap<>(), null);
+    }
+
+    @Override
+    protected void submitJob(List<WorkUnit> workUnits)
+        throws Exception {
+      this.workflowId = "someWorkflowId";
+    }
+  }
+
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    mockServiceStubs = mock(WorkflowServiceStubs.class);
+    mockClient = mock(WorkflowClient.class);
+    mockExecutionInfo = mock(WorkflowExecutionInfo.class);
+    DescribeWorkflowExecutionResponse mockResponse = mock(DescribeWorkflowExecutionResponse.class);
+    WorkflowServiceGrpc.WorkflowServiceBlockingStub mockBlockingStub = mock(WorkflowServiceGrpc.WorkflowServiceBlockingStub.class);
+    when(mockServiceStubs.blockingStub()).thenReturn(mockBlockingStub);
+    when(mockBlockingStub.describeWorkflowExecution(Mockito.any())).thenReturn(mockResponse);
+    when(mockResponse.getWorkflowExecutionInfo()).thenReturn(mockExecutionInfo);
+
+    mockWorkflowClientFactory = Mockito.mockStatic(TemporalWorkflowClientFactory.class);
+    mockWorkflowClientFactory.when(() -> TemporalWorkflowClientFactory.createServiceInstance(Mockito.anyString()))
+        .thenReturn(mockServiceStubs);
+    mockWorkflowClientFactory.when(() -> TemporalWorkflowClientFactory.createClientInstance(Mockito.any(), Mockito.anyString()))
+        .thenReturn(mockClient);
+
+    jobProperties = new Properties();
+    jobProperties.setProperty(ConfigurationKeys.FS_URI_KEY, "file:///");
+    jobProperties.setProperty(GobblinTemporalConfigurationKeys.TEMPORAL_CONNECTION_STRING, "someConnString");
+    jobProperties.setProperty(ConfigurationKeys.JOB_LOCK_TYPE, FileBasedJobLock.class.getName());
+    jobProperties.setProperty(ConfigurationKeys.SOURCE_CLASS_KEY, SimpleJsonSource.class.getName());
+  }
+
+  @BeforeMethod
+  public void methodSetUp() throws Exception {
+    mockStub = mock(WorkflowStub.class);
+    when(mockClient.newUntypedWorkflowStub(Mockito.anyString())).thenReturn(mockStub);
+    when(mockStub.getExecution()).thenReturn(WorkflowExecution.getDefaultInstance());
+
+    File tmpDir = Files.createTempDir();
+    String basePath = tmpDir.getAbsolutePath();
+    Path appWorkDir = new Path(basePath, "testAppWorkDir");
+    String jobLockDir = new Path(basePath, "jobLockDir").toString();
+    String stateStoreDir = new Path(basePath, "stateStoreDir").toString();
+    String jobName = "testJob";
+    String jobId = JobLauncherUtils.newJobId(jobName);
+    jobProperties.setProperty(ConfigurationKeys.JOB_NAME_KEY, jobName);
+    jobProperties.setProperty(ConfigurationKeys.JOB_ID_KEY, jobId);
+    jobProperties.setProperty(FileBasedJobLock.JOB_LOCK_DIR, jobLockDir);
+    jobProperties.setProperty(ConfigurationKeys.STATE_STORE_ROOT_DIR_KEY, stateStoreDir);
+
+    jobLauncher = new GobblinTemporalJobLauncherForTest(jobProperties, appWorkDir);
+  }
+
+  @Test
```

Comment:
add a test to check for workflow termination

Reply:
Done